pyomo/contrib/pynumero/interfaces/tests/test_nlp_projections.py

```diff
@@ -0,0 +1,195 @@
+        self.assertEqual(projected_nlp.primals_names(), expected_names)
+        np.testing.assert_equal(projected_nlp.get_primals(),np.asarray([1.0, 4.0, np.nan, 2.0]))
+        
+        self.assertTrue(np.array_equal(projected_nlp.evaluate_grad_objective(),
+                                       np.asarray([8.0, 9.0, 0.0, 1.0])))
+        self.assertEqual(projected_nlp.nnz_jacobian(), 5)
+        self.assertEqual(projected_nlp.nnz_hessian_lag(), 6)
+
+        J = projected_nlp.evaluate_jacobian()
+        self.assertEqual(len(J.data), 5)
+        denseJ = J.todense()
+        expected_jac = np.asarray([[6.0, 1.0, 0.0, 1.0],[1.0, 1.0, 0.0, 0.0]])
+        self.assertTrue(np.array_equal(denseJ, expected_jac))
+
+        # test the use of "out"
+        J = 0.0*J
+        projected_nlp.evaluate_jacobian(out=J)
+        denseJ = J.todense()
+        self.assertTrue(np.array_equal(denseJ, expected_jac))
+
+        H = projected_nlp.evaluate_hessian_lag()
+        self.assertEqual(len(H.data), 6)
+        expectedH = np.asarray([[2.0, 1.0, 0.0, 1.0],[1.0, 2.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0]])
+        denseH = H.todense()
+        self.assertTrue(np.array_equal(denseH, expectedH))
+
+        # test the use of "out"
+        H = 0.0*H
+        projected_nlp.evaluate_hessian_lag(out=H)
+        denseH = H.todense()
+        self.assertTrue(np.array_equal(denseH, expectedH))
+
+        # now test an expansion
+        projected_nlp = ProjectedNLP(nlp, ['x[0]', 'x[2]'])
+        expected_names = ['x[0]', 'x[2]']
+        self.assertEqual(projected_nlp.primals_names(), expected_names)
+        np.testing.assert_equal(projected_nlp.get_primals(),np.asarray([1.0, 4.0]))
+        
+        self.assertTrue(np.array_equal(projected_nlp.evaluate_grad_objective(),
+                                       np.asarray([8.0, 9.0])))
+        self.assertEqual(projected_nlp.nnz_jacobian(), 4)
+        self.assertEqual(projected_nlp.nnz_hessian_lag(), 4)
+
+        J = projected_nlp.evaluate_jacobian()
+        self.assertEqual(len(J.data), 4)
+        denseJ = J.todense()
+        expected_jac = np.asarray([[6.0, 1.0],[1.0, 1.0]])
+        self.assertTrue(np.array_equal(denseJ, expected_jac))
+
+        # test the use of "out"
+        J = 0.0*J
+        projected_nlp.evaluate_jacobian(out=J)
+        denseJ = J.todense()
+        self.assertTrue(np.array_equal(denseJ, expected_jac))
+
+        H = projected_nlp.evaluate_hessian_lag()
+        self.assertEqual(len(H.data), 4)
+        expectedH = np.asarray([[2.0, 1.0],[1.0, 2.0]])
+        denseH = H.todense()
+        self.assertTrue(np.array_equal(denseH, expectedH))
+
+        # test the use of "out"
+        H = 0.0*H
+        projected_nlp.evaluate_hessian_lag(out=H)
+        denseH = H.todense()
+        self.assertTrue(np.array_equal(denseH, expectedH))
+
+if __name__ == '__main__':
+    TestRenamedNLP().test_rename()
+    TestProjectedNLP().test_projected()
```

Comment:
Why are you calling out specific test cases and not just using
```python
if __name__ == '__main__':
    unittest.main()
```

Reply:
changed.